source/table_view.tsx

```diff
@@ -191,12 +191,17 @@ export class TableView extends React.Component<Properties, State> {
     }
   }
 
-  private onScrollHandler = () => {
-    const percent =
-      this.wrapperRef.current.scrollTop / this.wrapperRef.current.scrollHeight;
-    this.setState({topRow: Math.floor(percent * this.props.model.rowCount)});
+  private onScrollHandler = () => { 
+    const adjustedScrollTop = this.wrapperRef.current.scrollTop -
+      TableView.HEADER_HEIGHT;
+    const adjustedScrollHeight = this.wrapperRef.current.scrollHeight -
+      TableView.HEADER_HEIGHT
+    const percent = adjustedScrollTop / adjustedScrollHeight;
+    let topRow = Math.max(0, Math.floor(percent * this.props.model.rowCount));
+    this.setState({topRow: topRow});
   }
 
   private firstRowRef: React.RefObject<HTMLTableRowElement>;
   private wrapperRef: React.RefObject<HTMLDivElement>;
+  private static readonly HEADER_HEIGHT = 35;
```

Comment:
Order fields from independent to dependent. A read-only static field is as independent as it gets.